src/fstapp_server.erl

```diff
@@ -44,19 +44,20 @@ get_frequency() ->
 
 %% @hidden
 init([]) ->
+	{ok, Module} = application:get_env(fstapp, callback_module),
 	TimerRef = erlang:send_after(5000, self(), collect_metrics),
-	{ok, #state{timer_ref=TimerRef, freq=5000}}.
+	{ok, #state{timer_ref=TimerRef, freq=5000, callback_module=Module}}.
 
 %% @hidden
 handle_call(get_frequency, _From, State) ->
 	{reply, State#state.freq, State};
 handle_call(start_get_metrics, _From, State) ->
 	TimerRef = erlang:send_after(State#state.freq, self(), collect_metrics),
-	{reply, ok, #state{timer_ref=TimerRef, freq=State#state.freq}};
+	{reply, ok, #state{timer_ref=TimerRef, freq=State#state.freq, callback_module=State#state.callback_module}};
```

Comment:
this is super ugly. You have to update the state not building it again.
Here you only have to update the timer_ref